lib/inv_cg_quda.cpp

```diff
@@ -220,6 +220,8 @@ namespace quda {
 
   void CG::operator()(ColorSpinorField &x, ColorSpinorField &b, ColorSpinorField *p_init, double r2_old_init)
   {
+    if (param.is_preconditioner && param.global_reduction == false) commGlobalReductionSet(false);
```

Comment:
For true robustness, should this be a push/pop type operation for nested CG inversions?

Reply:
Yes.  The design for that goes beyond the scope of this pull.  It would be covered by #986.

Reply:
@maddyscientist